### ISSUE TITLE
how to setup gUserDefinedCPUList with only base frequency

### ISSUE THREAD

**Issue by hectorox** [Piker-Alpha/ssdtPRGen.sh#120]
hi,
This is my CPU:
http://ark.intel.com/products/39312/Intel-Core2-Duo-Processor-T9900-6M-Cache-3_06-GHz-1066-MHz-FSB?q=t9900

How should the User config.cfg look like, when the cpu only supports base frequency(also the cpu-type is Penryn)?
actually I set it up like that:
gUserDefinedCPUList=(
T9900,35,3060,3060,3060,2,2,(for now:9),266
)
Is this correct?

**Comment by Piker-Alpha** [comment]
Do you know the minimum frequency of your processor? And I hope that you have a power management SSDT taken from a Windows or Linux installation, which we can use to compare it with what the next update will generate.

**Comment by hectorox** [comment]
This is what ssdtPRGen says:
Override value: (-p) processor model, now using: T9900!

System information: Mac OS X 10.10.2 (14C109)
Brandstring 'Intel(R) Core(TM)2 Duo CPU T9900 @ 3.06GHz'

Scope (_PR_) {62 bytes} with ACPI Processor declarations found in the DSDT (ACPI 1.0 compliant)
Generating ssdt.dsl for a 'MacBookPro7,1' with board-id [Mac-F222BEC8]
Intel T9900 processor [0x1067A] setup [0x0301]
Processor matched!
With a maximum TDP of 35 Watt, as specified by Intel
Number logical CPU's: 2 (Core Frequency: 3060 MHz)
Number of Turbo States: 0
Number of P-States: 1 (3060-3060 MHz)
Injected C-States for P001 (C1,C3,C6)
Injected C-States for P002 (C1,C2,C3)
Warning: 'cpu-type' may be set improperly (0x0301 instead of 0x01)
Warning: 'board-id' [Mac-F222BEC8] is not supported by Unknown power management
Warning: 'system-type' may be set improperly (2 instead of 0)

Intel ACPI Component Architecture
ASL Optimizing Compiler version 20140926-64 [Nov  6 2014]
Copyright (c) 2000 - 2014 Intel Corporation

ASL Input:     /Users/hidekimotusawa/Library/ssdtPRGen/ssdt.dsl - 201 lines, 5310 bytes, 35 keywords
AML Output:    /Users/hidekimotusawa/Library/ssdtPRGen/ssdt.aml - 1000 bytes, 10 named objects, 25 executable opcodes

Compilation complete. 0 Errors, 0 Warnings, 0 Remarks, 1 Optimizations

I can get ssdt directly from bios, if you need. Or do you mean something else?

**Comment by hectorox** [comment]
I have an original SSDT dump with my CPU from a macbookpro. Can this help?

**Comment by Piker-Alpha** [comment]
Sure. You may e-mail that SSDT (see script for my e-mail address).

**Comment by hectorox** [comment]
This is strange because it was dumped directly from a macbookpro5,3 with linux. Is it possible that the _PSS Object is on a different table (I have a complete dump of all tables)?

(just for better understanding: Is the ssdt table a static table? Or is it a table that is always generated by itself reading cpu information?)

**Comment by Piker-Alpha** [comment]
Yes. The _PSS object is part of a dynamically loaded table. You may find on or more load() instructions for it in another SSDT table. And no it is not a static table, otherwise we could extract it from the firmware. You can probably find the memory address when you enable ACPI debug (see my blog for additional info.

This is the PerformanceStateArray of the MacBookPro5,3 that I extracted from the dump that I found:

<64 0a 00 00 00 7d 00 00 0a 00 00 00 0a 00 00 00 1d 0a 00 00 1d 0a 00 00>
<5a 09 00 00 66 6c 00 00 0a 00 00 00 0a 00 00 00 1c 09 00 00 1c 09 00 00>
<50 08 00 00 cc 5b 00 00 0a 00 00 00 0a 00 00 00 1a 08 00 00 1a 08 00 00>
<46 07 00 00 32 4b 00 00 0a 00 00 00 0a 00 00 00 19 07 00 00 19 07 00 00>
<3c 06 00 00 98 3a 00 00 0a 00 00 00 0a 00 00 00 17 06 00 00 17 06 00 00>

Translated to this ACPI object:

```Name (APSS, Package (0x05)
{
    Package (0x06) { 0x0a64, 0x7d00, 0x0a, 0x0a, 0x0a1d, 0x0a1d },
    Package (0x06) { 0x095a, 0x6c66, 0x0a, 0x0a, 0x091c, 0x091c },
    Package (0x06) { 0x0850, 0x5bcc, 0x0a, 0x0a, 0x081a, 0x081a },
    Package (0x06) { 0x074c, 0x4b32, 0x0a, 0x0a, 0x0719, 0x0719 },
    Package (0x06) { 0x063c, 0x3a98, 0x0a, 0x0a, 0x0617, 0x0617 }
})```

The problem is that 0x0a64 is only 2666 MHz and 0x7d00 is only 32000 (32 Watt) and thus this is not the data that you are looking for. Perhaps you need to add one extra P-State like this.

```Name (APSS, Package (0x06)
{
    Package (0x06) { 0x0bfa, 0x88b8, 0x0a, 0x0a, 0x0b1f, 0x0b1f },
    Package (0x06) { 0x0a64, 0x7d00, 0x0a, 0x0a, 0x0a1d, 0x0a1d },
    Package (0x06) { 0x095a, 0x6c66, 0x0a, 0x0a, 0x091c, 0x091c },
    Package (0x06) { 0x0850, 0x5bcc, 0x0a, 0x0a, 0x081a, 0x081a },
    Package (0x06) { 0x074c, 0x4b32, 0x0a, 0x0a, 0x0719, 0x0719 },
    Package (0x06) { 0x063c, 0x3a98, 0x0a, 0x0a, 0x0617, 0x0617 }
})```

**Comment by Piker-Alpha** [comment]
I received your Windows extracted ACPI SSDT's and the _PSS object in that table looks like this:
```
Name (_PSS, Package (0x05)
{
    Package (0x06) { 0x0BFC, 0x88B8, 0x0A, 0x0A, 0x0C2C, 0x0C2C }, 
    Package (0x06) { 0x0BFB, 0x88B8, 0x0A, 0x0A, 0x4B26, 0x4B26 }, 
    Package (0x06) { 0x0855, 0x55F0, 0x0A, 0x0A, 0x081D, 0x081D },
    Package (0x06) { 0x0640, 0x3E80, 0x0A, 0x0A, 0x0617, 0x0617 }, 
    Package (0x06) { 0x0320, 0x36B0, 0x0A, 0x0A, 0x860F, 0x860F }

})
```

**Comment by hectorox** [comment]
Hi Piker-Alpha,

Did you ever added support for this above CPU Model T9900?
